bin/up

```diff
@@ -36,17 +36,34 @@ function check_config() {
 }
 
 function check_sharelatex_env_vars() {
+  set +o pipefail
+  local rc_occurrences=$(grep -o SHARELATEX_ "$TOOLKIT_ROOT/config/overleaf.rc" | wc -l | sed 's/ //g')
+  set -o pipefail
+
+  if [ "$rc_occurrences" -gt 0 ]; then
+    echo "WARNING: some runtime variables defined in config/overleaf.rc" 
+    echo "contain 'SHARELATEX_' in their name. The variables should be"
+    echo "renamed to 'OVERLEAF_'."
+    echo "you can upgrade your config/overleaf.rc by running bin/rename-rc-vars"
+    prompt "Do you want to continue?"
+  fi
+
   local invalid_prefix="SHARELATEX_"
   if [[ "$IMAGE_VERSION_MAJOR" -lt 5 ]]; then
     invalid_prefix="OVERLEAF_"
   fi
 
-  if grep -q "$invalid_prefix" "$TOOLKIT_ROOT/config/variables.env"; then
+  set +o pipefail
+  local env_occurrences=$(grep -o "$invalid_prefix" "$TOOLKIT_ROOT/config/variables.env" | wc -l | sed 's/ //g')
+  set -o pipefail
+
+  if [ "$env_occurrences" -gt 0 ]; then
```

Comment:
Just curious, was there anything "wrong" with the previous exit code based check to identify if any lines need migrating?

Reply:
Yeah, here's an example:

With `pipefail`:

```bash
#! /usr/bin/env bash

set -euo pipefail

touch __tmp

count=$(grep -o SOMETHING __tmp | wc -l | sed 's/ //g')

echo "found $count occurrences"
```

``` bash
> ./example # empty response
> echo $?
1
```

Without `pipefail`:

```bash
#! /usr/bin/env bash

set -euo pipefail

touch __tmp

set +o pipefail
count=$(grep -o SOMETHING __tmp | wc -l | sed 's/ //g')
set +o pipefail

echo "found $count occurrences"
```

``` bash
> ./example # empty response
found 0 occurrences
> echo $?
0
```

Reply:
Right, for the case where you want to know the number of entries we need a bit of larger snippet. But here we only care about "has matching entries" (`grep -q` -> exit 0) or "does not have any matching entries" (`grep -q` -> exit 1). 

Btw, `grep --count` emits `0`, `1` or any other number of entries it found, which might be "cleaner" :) You would still need to handle the non-zero exit code for the case where zero entries were found.

```bash
jackermann@x1:/tmp$  cat foo.log 
foo
jackermann@x1:/tmp$  grep -e foo --count foo.log 
1
jackermann@x1:/tmp$  echo $?
0
jackermann@x1:/tmp$  grep -e bar --count foo.log 
0
jackermann@x1:/tmp$  echo $?
1
```